app/code/core/Mage/Core/functions.php

```diff
@@ -123,6 +123,10 @@ function mageCoreErrorHandler($errno, $errstr, $errfile, $errline)
         return false;
     }
 
+    if (isset($_ENV['DEV_PHP_STRICT']) && $_ENV['DEV_PHP_STRICT'] == '1') {
+        return false;
+    }
+
     // Suppress deprecation warnings on PHP 7.x
     if ($errno == E_DEPRECATED && version_compare(PHP_VERSION, '7.0.0', '>=')) {
         return true;
```

Comment:
What is the purpose of this variable exactly? The workaround in the linked PRs was meant to allow throwing deprecation errors, but that's only achieved by ignoring the below condition.

Reply:
Set `DEV_PHP_STRICT` to 1 has the same effect as comment the lines (every time)